### DISCUSSION TITLE
ntp transistor not working

### DISCUSSION THREAD

**Discussion by sensormatic1985** [taligentx/dscKeybusInterface#230] [dscKeybusInterface Discussion]
good afternoon
I have made the connection of the transistor ntp-3904 but at the moment of sending the assembly, my hard work lost the connection to the internet and after a few seconds it is online again, but it does not arm or disarm, someone could help me with this

**Comment by kricon** [comment]
Try KeybusReader or Status sketch, connected with USB on PC and try sending keys onto serial monitor, to verify that transistor is working - sending keys and that DSC panel is receiving them.
It's not NTP, but it's an NPN 2N3904 transistor. Are you sure that you've wired it correctly? It have pinout exactly opposite of BC547.
If it doesn't work, there is possibility that transisor is blown, its easy to burn the transistor if you connected it wrong or miscalculated the base resistor.

**Reply by sensormatic1985** [reply]
If you connect it as stated in the connection guide that appears in the code
emitter to terminal - from dsc panel
base + 1k resistor to audino port
collector to the grenn terminal of the D.S.C data bus

**Reply by kricon** [reply]
Yes, that is correct. Emitter is first pin from the left when you're looking on text etched to transistor itself.
If it doesn't work with KeybusReader/Status sketch, and all wiring is check good, then its bad transistor.

**Reply by sensormatic1985** [reply]
How do I test it with keybusreader?

I already changed the transistor for a new one, also I already loaded the keybusreader code on my arduino

**Reply by kricon** [reply]
In Arduino IDE, go to tools - serial monitor. Verify that you're getting output from panel to the serial monitor (try to press something or real keypad or open/close zones), if you get garbage then change baud rate.
For seding keys into serial monitor, just write and press enter or send, you have list of all keys in readme.md (search for Virtual keypad). You can, for example, type 4 digits, press send and you should see that panel accepted those keys (even hear beeps from physical dsc keypad).

**Reply by sensormatic1985** [reply]
I am using an arduno uno with an internet lan card, I can do the tests that you say or I have to change it for an EPS8266 card

**Reply by kricon** [reply]
You can do all the test with just an Arduino.

**Reply by sensormatic1985** [reply]
ok I already did the test but I do not have the factory key 1234 and then I send it and nothing happens, you have to configure something to the programming of the DSC panel

**Reply by kricon** [reply]
No, you don't need to configure anything in DSC panel programming. It should work with any keys.
Do you even see any output in serial monitor with keybusreader sketch?
But when you send any keys, or even sending "s" (for stay arming), you dont get anything at all?
If so - then transistor connection to the panel (or arduino) is bad, or it's broken transistor.
Can you upload the picture of your wiring, maybe it will be easier to help it that way.

**Comment by sensormatic1985** [comment]
https://drive.google.com/file/d/1VDwcnD7FCK__rSXVG2TIxw9TWwgvADvR/view?usp=sharing
https://drive.google.com/file/d/1t-7yBNLJRnLtBHP_7k35WeSnLK4eZl84/view?usp=sharing

**Reply by sensormatic1985** [reply]
https://drive.google.com/file/d/13WbyBc9xYKMgF74SU3mzez948Vr-lvTA/view?usp=sharing

**Reply by kricon** [reply]
I bet that problem is with that cheapo protoboard. Just twist the wires on transistor legs and connect it to the screwterminals, without using dupont wires and protoboard.

**Reply by sensormatic1985** [reply]
I performed a previous test by soldering the wires to the transistor with the resistor
brown cable -negative
arduino pin blue wire
Green data cable Alarm panel
attached photo link
https://drive.google.com/file/d/12g6I71SN2RllQgTpxDjfQUYattum7jMI/view?usp=sharing

**Reply by kricon** [reply]
Yes, that is correct pinout for 2n3904.
Only other possible thing, is that ethernet shield is somehow interfering with Arduino output/input pins, can you temporary try without it?

**Reply by sensormatic1985** [reply]
okn I'm going to do the tests, give me a few minutes and I'll tell you how it went

**Reply by sensormatic1985** [reply]
I already did the tests but I have not been successful
Attached link of photos of the tests

https://drive.google.com/file/d/12pPC9kWw0gVeQcvqK_-32lwBWOGPi-ko/view?usp=sharing
https://drive.google.com/file/d/11T9yqHs1KG1KPLRWP8D2w-gD8fxjCLQj/view?usp=sharing

**Reply by kricon** [reply]
I don't have any other ideas. Maybe it's just an bad Arduino output, which is unlikely but you can try changing pin defined for dscwrite. If transistor is good and properly connected, it should work. BTW: I suppose you pressed keys "1", "2" and get "invalid access code" error from physical DSC keypad? Same thing should be if you enter code in serial monitor and send it.

**Reply by sensormatic1985** [reply]
good morning
If I change the pin but anyway, I do not know why they do not work for me, thanks for your help, you are very kind, one last question this arduino code could be used as a communicator contac id- for monitoring centers, or it is only designed for self monitoring

**Reply by kricon** [reply]
It is intended for self monitoring.

**Reply by sensormatic1985** [reply]
Is there any way to make a project to be cheaper than a contac id communicator that is currently on the market
Suddenly as a business idea, I don't know what you think

**Reply by kricon** [reply]
I don't think that monitoring company will accept any homemade-like project like this one. It's purely intended for self monitoring and home automation integration done by yourself and at your own risk.

**Reply by sensormatic1985** [reply]
ok friend thank you very much I will continue trying with my npn transistor if it works for me, because I already have my home assistant working, I receive the signals of my alarm but I have not been able to arm or disarm remotely
